packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_control_pane.dart

```diff
@@ -45,6 +60,36 @@ class SnapshotControlPane extends StatelessWidget {
   }
 }
 
+class _ClassFilter extends StatelessWidget {
+  const _ClassFilter({Key? key, required this.onChanged}) : super(key: key);
+
+  final Function(String value) onChanged;
+
+  @override
+  Widget build(BuildContext context) => DevToolsClearableTextField(
+        labelText: 'Class Filter',
+        hintText: 'Filter by class name',
+        onChanged: onChanged,
+      );
+}
+
+class _ToCsv extends StatelessWidget {
+  const _ToCsv({Key? key, required this.item}) : super(key: key);
+
+  final SnapshotInstanceItem item;
+
+  @override
+  Widget build(BuildContext context) {
+    return IconLabelButton(
```

Comment:
make the download to CSV button a button in common_widgets so it can be reused by `_ExportAllocationProfileButton`. In general though, why do we want users to be downloading CSV files instead of downloading DevTools exports that can be reloaded in DevTools?

Reply:
made

users may want to analyze data deeper in google sheets
we will never support all google sheet features :)